server/src/main/java/org/opensearch/wlm/listeners/WorkloadManagementRequestFailureListener.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.wlm.listeners;
+
+import org.opensearch.action.search.SearchPhaseContext;
+import org.opensearch.action.search.SearchRequestContext;
+import org.opensearch.action.search.SearchRequestOperationsListener;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.wlm.QueryGroupService;
+import org.opensearch.wlm.QueryGroupTask;
+
+/**
+ * Request listener to listen for failures and update query group failure stats
+ */
+public class WorkloadManagementRequestFailureListener extends SearchRequestOperationsListener {
```

Comment:
Can we reuse `QueryGroupRequestRejectionOperationListener` for this? We can rename it to something more generic

Reply:
Yes! that is a good callout. I will merge it with rejection listener and appropriately rename it